dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumMonitor.kt

```diff
@@ -0,0 +1,149 @@
+     * @param attributes additional custom attributes to attach to the view
+     * @see [startView]
+     */
+    fun stopView(
+        key: Any,
+        name: String,
+        attributes: Map<String, Any?> = emptyMap()
+    )
+
+    /**
+     * Notify that a new Resource is being loaded, linked with the [key] instance.
+     * @param key the instance that represents the resource being loaded (usually your
+     * request or network call instance).
+     * @param url the url or local path of the resource being loaded
+     * @param attributes additional custom attributes to attach to the resource
+     * @see [stopResource]
+     * @see [stopResourceWithError]
+     */
+    fun startResource(
+        key: Any,
+        url: String,
+        attributes: Map<String, Any?> = emptyMap()
+    )
+
+    /**
+     * Stops a previously started Resource, linked with the [key] instance.
+     * @param key the instance that represents the active view (usually your
+     * request or network call instance).
+     * @param mimeType the (nullable) mime type of the loaded resource, used to categorize it
+     * @param attributes additional custom attributes to attach to the resource
+     * @see [startResource]
+     * @see [stopResourceWithError]
+     */
+    fun stopResource(
+        key: Any,
+        mimeType: String?,
+        attributes: Map<String, Any?> = emptyMap()
+    )
+
+    /**
+     * Stops a previously started Resource that failed loading, linked with the [key] instance.
+     * @param key the instance that represents the active view (usually your
+     * request or network call instance).
+     * @param message a message explaining the error
+     * @param origin the origin of the error (eg: "network", "sqlight", "assets", …)
+     * @param throwable the throwable
+     * @see [startResource]
+     * @see [stopResource]
+     */
+    fun stopResourceWithError(
+        key: Any,
+        message: String,
+        origin: String,
+        throwable: Throwable
+    )
+
+    /**
+     * Notifies that an error occurred in the active View.
+     * @param message a message explaining the error
+     * @param origin the origin of the error (eg: "network", "sqlight", "assets", …)
+     * @param throwable the throwable
+     * @param attributes additional custom attributes to attach to the error
+     */
+    fun addError(
+        message: String,
+        origin: String,
+        throwable: Throwable,
+        attributes: Map<String, Any?>
+    )
+
+    // region Builder
+
+    /**
+     * A Builder class for a [RumMonitor].
+     */
+    class Builder {
+
+        /**
+         * Builds a [RumMonitor] based on the current state of this Builder.
+         */
+        fun build(): RumMonitor {
+            return if (!RumFeature.isInitialized()) {
+                devLogger.e(Datadog.MESSAGE_NOT_INITIALIZED)
+                NoOpRumMonitor()
+            } else {
+                GlobalRumMonitor.updateApplicationId(RumFeature.applicationId)
```

Comment:
maybe we should call this from RumFeature#initialize method, I feel it doesn't belong here in this context.

Reply:
You're right will do